app.ts

```diff
@@ -0,0 +1,295 @@
+window.onload = async function () {
+  $("#loader").hide();
+  await fetchTotalRecordCount();
+  updateRecordsPerPage();
+  fetchAndDisplayColumns();
+  handleSpecialResize();
+};
+// Global variables
+const IMQS: string = "http://localhost:2050";
+let totalRecordCount: number;
+let totalPages: number;
+let currentValueOfFirstRecord: number = 1;
+let currentFirstRecordIndex: number = 1;
+let currentPage: number = 1;
+let recordsPerPage: number;
+let resizeTimeout: number | undefined;
+let searchResultDisplay = false;
+let isSearchMode = false;
+let searchResultIndexes: number[] = [];
+let specialResizeOccurred = false;
```

Comment:
Try sticking to two global variables. And if you set a variable no need to give the type aswell